packages/jest/src/index.js

```diff
@@ -86,48 +12,31 @@ module.exports = (neutrino, opts = {}) => {
         'jest/no-identical-title': 'error',
         'jest/valid-expect': 'error'
       }
-    }));
-
-  neutrino.on('test', (args) => {
-    const usingBabel = neutrino.config.module.rules.has('compile');
-
-    neutrino.config.when(usingBabel, () => {
-      neutrino.use(loaderMerge('compile', 'babel'), {
-        retainLines: true,
-        plugins: [
-          // Once babel-preset-jest has better Babel 7 support we should switch back to it
-          // (or even use babel-jest, which will allow simplifying the transformer too):
-          // https://github.com/facebook/jest/issues/6126
-          // For now this plugin is taken from here (we don't need object-rest-spread since node >=8.3):
-          // https://github.com/facebook/jest/blob/v22.4.2/packages/babel-preset-jest/index.js#L11-L12
-          require.resolve('babel-plugin-jest-hoist'),
-          // Since the tests will be run by node which doesn't yet support ES2015 modules
-          require.resolve('@babel/plugin-transform-modules-commonjs')
-        ]
-      });
     });
-
-    // We need to parse argv separately in order to identify files
-    // and jest-related options since root neutrino does not know about
-    // jest options and will provide wrong/incomplete `args.files`
-    const jestArgs = yargs
-      .command('test [files..]', 'Run tests', jestOptions)
-      .argv;
-    const configFile = join(tmpdir(), 'config.json');
-    const options = normalizeJestOptions(opts, neutrino, usingBabel);
-    const cliOptions = {
-      ...jestArgs,
-      // Jest is looking for Array of files in `argv._`. Providing them
-      _: jestArgs.files,
-      config: configFile,
-      coverage: args.coverage,
-      watch: args.watch
-    };
-
-    writeFileSync(configFile, `${JSON.stringify(options, null, 2)}\n`);
-
-    return jest
-      .runCLI(cliOptions, options.roots || [options.rootDir])
-      .then(({ results }) => results.success ? Promise.resolve() : Promise.reject());
   });
+
+  neutrino.config.when(
+    neutrino.config.module.rules.has('compile') &&
+    neutrino.options.command === 'jest',
+    () => {
```

Comment:
Given that the false case isn't used, and we're not using a passed `(config)`, this might be cleaner as:
```js
  if (
    neutrino.config.module.rules.has("compile") &&
    neutrino.options.command === "jest"
  ) {
    // ...
  }
```